File: dynamic_array_heap.py
import sys

def main():
    arr = []
    print("Enter elements (enter -1 to stop):")
    
    while True:
        try:
            val = int(input())
            if val == -1:
                break
            arr.append(val)
        except ValueError:
            break

    print("Dynamic Heap Array Contents:")
    for i, val in enumerate(arr):
        print(f"arr[{i}] = {val}")
    
    print(f"Memory usage: {sys.getsizeof(arr)} bytes")

if __name__ == "__main__":
    main()

File: DynamicArrayHeap.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    vector<int> arr;
    cout << "Enter elements (enter -1 to stop):\n";
    
    int val;
    while (cin >> val && val != -1) {
        arr.push_back(val);
    }

    cout << "Dynamic Heap Array Contents:\n";
    for (size_t i = 0; i < arr.size(); i++) {
        cout << "arr[" << i << "] = " << arr[i] << endl;
    }

    cout << "Current capacity: " << arr.capacity() << endl;
    return 0;
}